lib/Sema/SemaDecl.cpp

```diff
@@ -11262,6 +11270,100 @@ void Sema::AddInitializerToDecl(Decl *RealDecl, Expr *Init, bool DirectInit,
   CheckCompleteVariableDeclaration(VDecl);
 }
 
+/// Checked C: Validate that the NT_CHECKED type array initializers must be null terminated
+bool Sema::ValidateNTCheckedType(ASTContext &Ctx, QualType VDeclType, Expr *Init) {
+  if(!Init) {
+    return true;
+  }
+  const Type *current = VDeclType.getTypePtr();
+  switch (current->getTypeClass()) {
+    case Type::Pointer: {
+      // Pointers to NT_CHECKED types can be initialized by any Expr option (e.g., calling a function)
+      // Validating initializers of pointers require further analysis
```

Comment:
This comment is not quite accurate.    Epressions with type nt_array_ptr point to valid null-terminated values.   This is enforced static by type checking and checking of casts.  We don't need to check further here.